nixos/modules/services/audio/jmusicbot.nix

```diff
@@ -0,0 +1,39 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.jmusicbot;
+in
+{
+  options = {
+    services.jmusicbot = {
+      enable = mkEnableOption "jmusicbot, a Discord music bot that's easy to set up and run yourself";
+
+      stateDir = mkOption {
+        type = types.path;
+        description = ''
+          Location where config.txt and serversettings.json is saved.
+        '';
+        default = "/var/lib/jmusicbot/";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.jmusicbot = {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" ];
+      description = "Discord music bot that's easy to set up and run yourself!";
+      serviceConfig = {
+        ExecStart = "${pkgs.jmusicbot}/bin/JMusicBot";
+        WorkingDirectory = cfg.stateDir;
+        Restart = "always";
+        RestartSec = 20;
+        DynamicUser = true;
+        StateDirectory = [ name "${cfg.stateDir}" ];
```

Comment:
Sorry I didn't explain this well enough. `StateDirectory` won't accept absolute paths. You pass `StateDirectory` a directory name and this will be created in `/var/lib`.

I generally suggest conditionally provisioning the state directory only if the default value is set (like we do for users).

```
serviceConfig = mkMerge [
  {
    ExecStart = ...
  }
  (mkIf (cfg.stateDir == "/var/lib/jmusicbot") { StateDirectory = "jmusicbot"; })
];
```

Then you can set the `description` to this:
```
The directory where config.txt and serversettings.json is saved. If left as the default value this directory will automatically be created before the JMusicBot server starts, otherwise the sysadmin is responsible for ensuring the directory exists with appropriate ownership and permissions. 
```

Let me know if you want further clarification.

Reply:
Thanks. I think I should have it right now.

Reply:
Perfect!